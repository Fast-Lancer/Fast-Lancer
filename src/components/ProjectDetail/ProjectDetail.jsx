import { Link } from 'react-router-dom'
import styles from './ProjectDetail.module.css'
import clock from '../../assets/icons/clockSmall.png'
import www from '../../assets/icons/wwwSmall.png'
import money from '../../assets/icons/moneySmall.png'

export default function ProjectDetail({ project }) {
  const {
    hourly_rate,
    URL,
    date_start,
    date_end,
    description,
    notes,
    clients,
    hours_worked,
    price_quoted,
    hours_quoted,
  } = project

  return (
    <main>
      <div className={styles.projectDetail}>
        <div className={styles.projectTitle}>{project.title}</div>
        <div className={styles.clientName}>
          <Link to={`/clients/${project.client_id}`}>
            <h1 aria-label="client-name">
              {clients?.client_name ?? 'No client name entered'}
            </h1>
          </Link>
          <h3>Client</h3>
        </div>
        <div className={styles.infoWrapper}>
          <h3>{URL && (
            <div className={styles.infoItem} aria-label="URL">
              <Link to={`${URL}`}><img src={www} alt="www" />
                {URL}
              </Link>
            </div>
          )}</h3>
          <section className={styles.infoSection}>          
            <div className={styles.infoItem} aria-label="hourly-rate">
              <img src={money} alt="money" />
            Hourly Rate: ${hourly_rate}
            </div>
            <div className={styles.infoItem} aria-label="start-date">
              <img src={clock} alt="clock" />
            Started: {date_start}
            </div>

            {hours_worked && (
              <div className={styles.infoItem} aria-label="hours-logged">
                <img src={money} alt="money" />
            Hours Logged: {hours_worked}
              </div>
            )}
            <div className={styles.infoItem} aria-label="end-date">
              <img src={clock} alt="clock" />
            Deadline: {date_end}
            </div>
          </section>
        </div>
        <section className={styles.alignLeft}>
          <h2>Description</h2>
          <p className={styles.textBlock} aria-label="description">
            {description ? description : 'No description entered.'}
          </p>
          <h2>Notes</h2>
          <p className={styles.textBlock} aria-label="notes">
            {notes ? notes : 'No notes entered.'}
          </p>
        </section>
      </div>
    </main>
  )
}
