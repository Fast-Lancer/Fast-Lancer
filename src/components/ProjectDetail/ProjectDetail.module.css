.projectDetail {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  background-color: #d3d3d3;
  font-size: 1.2em;
  width:90%;
}

.projectDetail > * {
  padding: 10px;
}

h2 {
  padding:5px;
}

.infoSection {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  font-size: 1.2em;
  padding-bottom:1.2em;
}

.infoItem{
  display: flex;
  justify-content: center;
  align-items: center;
}

.infoItem img{
  padding: 5px;
}

.clientName {
  width: 100%;
  height: fit-content;
  display: flex;
  justify-content: center;
  border-bottom: 1px solid black;  

  font-family:'IBM Plex Mono', monospace;
}

.clientName h1 {
  font-family:'IBM Plex Mono', monospace;
  font-size: 2em;
}

.clientName > * {
  margin: 0;
}

.alignLeft {
  align-self: center;
  border-bottom: 1px black solid;
  width:100%;
  padding: 10px;
}


.alignLeft > p {
  margin-left: 20px;
  display: flex;
  align-items: center;
}

.alignLeft:last-child{
  border:none;
}

.alignItem{
  display: flex;
  align-items: center;
  padding: 10px;
  padding-top: 10px;
  font-size: 1.1em;
}

.alignItem img{
  padding: 5px;
}

.textBlock{
  padding: 10px;
  font-family:'IBM Plex Sans', monospace;
}

@media(max-width: 768px) {
  .infoSection {
    flex-direction: column;
  }

  .projectDetail {
    align-items: center;
  }
}