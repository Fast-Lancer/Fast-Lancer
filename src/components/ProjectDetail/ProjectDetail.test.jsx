import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import ProjectDetail from './ProjectDetail.jsx'

it('should display project detail with provided content', () => {
  const { container } = render(
    <MemoryRouter>
      <ProjectDetail project={{ hourly_rate: 33, URL: 'something', date_start: '2022-01-20', date_end: '2022-01-20', description: 'something', notes: 'something', clients: { name: 'person' }  }}/>
    </MemoryRouter>
  )

  expect(container).toMatchSnapshot()
})
