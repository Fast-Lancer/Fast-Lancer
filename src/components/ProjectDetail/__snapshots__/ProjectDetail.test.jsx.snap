// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display project detail with provided content 1`] = `
<div>
  <main
    class="projectDetail"
  >
    <div
      class="clientName"
    >
      <h1
        aria-label="client-name"
      >
        No client name entered
      </h1>
    </div>
    <section
      class="infoSection"
    >
      <div
        aria-label="hourly-rate"
      >
        Hourly Rate: 
        33
      </div>
      <div
        aria-label="URL"
      >
        URL: 
        something
      </div>
      <div
        aria-label="start-date"
      >
        Start Date: 
        2022-01-20
      </div>
      <div
        aria-label="end-date"
      >
        End Date: 
        2022-01-20
      </div>
    </section>
    <section
      class="alignLeft"
    >
      <h2>
        Description
      </h2>
      <p
        aria-label="description"
      >
        something
      </p>
    </section>
    <section
      class="alignLeft"
    >
      <h2>
        Notes
      </h2>
      <p
        aria-label="notes"
      >
        something
      </p>
    </section>
    <section
      class="alignLeft"
    >
      <h2>
        Hours and Pricing
      </h2>
      <p>
        Hours Worked: 
      </p>
      <p>
        Hours Quoted: 
      </p>
      <p>
        Price Quoted: 
      </p>
    </section>
  </main>
</div>
`;
